src/components/teamBoxes/TeamOneBox.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import "./redTeamBox.css";
+import { get, ref, set, child, onValue } from "firebase/database";
+import { database } from "../../utils/firebase";
+import { setTeamOneOperatives, setTeamOneSpymaster } from "../../store/teamOneSlice";
+
+const RedTeamBox = () => {
+  let playerId = useSelector((state) => state.player.playerId);
+  let playerRef;
+  const roomId = useSelector((state) => state.player.roomId);
+  const username = useSelector((state) => state.player.username);
+  const dispatch = useDispatch();
+  playerRef = ref(database, "players/" + playerId);
+  const teamOneRef = ref(database, `rooms/${roomId}/team-1/`);
+  const teamTwoRef = ref(database, `rooms/${roomId}/team-2/`);
+  const teamOneOperativesRef = ref(database, `rooms/${roomId}/team-1/operatives/`);
+  const teamOneSpymasterRef = ref(database, `rooms/${roomId}/team-1/spymaster/`);
+  const { teamOneOperatives } = useSelector(state => state.teamOne);
+  const { teamOneSpymaster } = useSelector(state => state.teamOne);
```

Comment:
these be combined in one line, right? I usually write my useSelectors a bit differently so i'm not sure!

Reply:
Not sure where I picked this up but its just deconstructing the "teamOneOperatives" out of state.teamOne! So its the same as "const teamOneOperatives = useSelector(state => state.teamOne.teamOneOperatives)

Reply:
ohhhh i see. gotcha makes sense.